src/modules/fw_rate_control/FixedwingRateControl.cpp

```diff
@@ -280,13 +280,51 @@ void FixedwingRateControl::Run()
 
 			// update saturation status from control allocation feedback
 			control_allocator_status_s control_allocator_status;
+			control_allocator_status_s control_allocator_status_diff_thr;
+
+			// VTOL differential thrust feature: the motors are in matrix 0, and if differential thrust is enabled
+			// assume that it's the only means of control around this axis and thus only check the matrix 0 saturation
+			// around the corresponding axis.
+			Vector<bool, 3> diff_thr_en;
+			diff_thr_en(0) = (_vehicle_status.is_vtol_tailsitter
+					  && _param_vt_fw_difthr_en.get() & static_cast<int32_t>(VtFwDifthrEnBits::YAW_BIT))
+					 || (!_vehicle_status.is_vtol_tailsitter
+					     && _param_vt_fw_difthr_en.get() & static_cast<int32_t>(VtFwDifthrEnBits::ROLL_BIT));
+			diff_thr_en(1) = _param_vt_fw_difthr_en.get() & static_cast<int32_t>(VtFwDifthrEnBits::PITCH_BIT);
+			diff_thr_en(2) = (_vehicle_status.is_vtol_tailsitter
+					  && _param_vt_fw_difthr_en.get() & static_cast<int32_t>(VtFwDifthrEnBits::ROLL_BIT))
+					 || (!_vehicle_status.is_vtol_tailsitter
+					     && _param_vt_fw_difthr_en.get() & static_cast<int32_t>(VtFwDifthrEnBits::YAW_BIT));
+
+			// saturation handling for axis controlled by differential thrust (VTOL only)
+			if (_param_vt_fw_difthr_en.get() > 0
+			    && _control_allocator_status_subs[0].update(&control_allocator_status_diff_thr)) {
+				Vector<bool, 3> saturation_positive;
+				Vector<bool, 3> saturation_negative;
+
+				for (size_t i = 0; i < 3; i++) {
+					if (diff_thr_en(i)) {
+						if (control_allocator_status.unallocated_torque[i] > FLT_EPSILON) {
+							saturation_positive(i) = true;
+
+						} else if (control_allocator_status.unallocated_torque[i] < -FLT_EPSILON) {
+							saturation_negative(i) = true;
+						}
+					}
+				}
+
+				// TODO: send the unallocated value directly for better anti-windup
+				_rate_control.setSaturationStatus(saturation_positive, saturation_negative);
```

Comment:
Aren't all the saturation flags overwritten in the `_rate_control.setSaturationStatus()` call from the other matrix below 🤔

Reply:
yes :grimacing: 
Guess that's what reviews are for:)
I think it's fine if we fill the same flags for both matrices and then apply them once to the controller.